src/components/Transfer.tsx

```diff
@@ -47,59 +49,60 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function Transfer() {
+  const { dispatchTransaction } = useUpdateTransactionContext();
   const classes = useStyles();
+  const [input, setInput] = useState<string>('0');
   const [isRunning, setIsRunning] = useState(false);
-  const [helperText, setHelperText] = useState('');
-  const [transferInput, setTransferInput] = useState<string>('');
-  const [actualInput, setActualInput] = useState<number | null>();
   const [amountNotCorrect, setAmountNotCorrect] = useState<boolean>(false);
   const { sourceChainDetails, targetChainDetails } = useSourceTarget();
   const { account } = useAccounts();
 
-  const planck = 10 ** targetChainDetails.apiConnection.api.registry.chainDecimals[0];
-  const { estimatedFee, receiverAddress, estimatedFeeLoading } = useTransactionContext();
-  const { api, isApiReady } = sourceChainDetails.apiConnection;
+  const {
+    estimatedFee,
+    transferAmount,
+    transferAmountError,
+    receiverAddress,
+    estimatedFeeLoading
+  } = useTransactionContext();
+  const { api } = sourceChainDetails.apiConnection;
   const balance = useBalance(api, account?.address || '');
 
   const { isButtonDisabled, sendLaneMessage } = useSendMessage({
-    input: actualInput?.toString() ?? '',
+    input: transferAmount?.toString() ?? '',
     isRunning,
     setIsRunning,
     type: TransactionTypes.TRANSFER
   });
 
   const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target.value) {
-      const [actualValue, message] = evalUnits(event.target.value);
-      setHelperText(message);
-      setActualInput(actualValue && actualValue * planck);
-    }
-    setTransferInput(event.target.value);
+    const actualValue = event.target.value;
+    setInput(actualValue);
+    dispatchTransaction(
+      TransactionActionCreators.setTransferAmount(
+        actualValue !== null ? actualValue?.toString() : '',
+        api.registry.chainDecimals[0]
+      )
+    );
   };
 
-  useEffect((): void => {
-    isRunning && setTransferInput('');
-  }, [isRunning]);
-
-  // To extract estimated fee logic to specific component. Issue #171
   useEffect((): void => {
     estimatedFee &&
-      actualInput &&
-      setAmountNotCorrect(new BN(balance.free).sub(new BN(actualInput).add(new BN(estimatedFee))).toNumber() < 0);
-  }, [actualInput, estimatedFee, balance, isApiReady]);
+      transferAmount &&
+      setAmountNotCorrect(new BN(balance.free).sub(transferAmount).add(new BN(estimatedFee)).isNeg());
+  }, [transferAmount, estimatedFee, api.registry.chainDecimals, balance.free]);
```

Comment:
can't we move this to the reducer as well? if we can achieve this we would be pretty close to the presentational component approach we are looking for components.

Also can't this message be included in the `helperText` ?